dataService.js

```diff
@@ -0,0 +1,94 @@
+// Maps
+let productMap;
+let categoryMap;
+let combinedProductMap;
```

Comment:
I've removed explicit initialization with `null` values as default `undefined` is treated the same way in JS, i.e. `falsely` value.

Read more [here](https://codeburst.io/javascript-null-vs-undefined-20f955215a2)